### PR TITLE
 👌 IMPROVE & 🐛 FIX

### PR DESCRIPTION
**Pull request by mhmdio**
**This PR includes the following improvements:**

- pump TF version to 0.13.5
- pump AWS TF provider version to 3.15.0
- add kms_alias_name as an optional variable 

**And also, includes the following fixes:**

- fix unneeded region variables
- fix wrong bucket module source
- fix racing condition between S3 bucket policy and S3 public access policy

### CODE REVIEW

**Comment by sblack4** [comment]
Thank you! This looks great to me, the only concern I have is using the switch to using the provider's region rather than specifying the region. @cdaniluk @smiller171 do you want to weigh in?

File: examples/basic/provider.tf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+provider "aws" {
+  region = "eu-central-1"
+}
```
Comment (sblack4) [inline examples/basic/provider.tf]: I like to see a newline at the end of files but I'm not blocking on that minutia

**Review by sblack4 — APPROVED** [review]

File: examples/basic/provider.tf (hunk 1 above)
Comment (mhmdio) [reply examples/basic/provider.tf]: Thanks for the quick replay, tf fmt did this maybe, using it with vscode, but seems not working properly.

**Comment by mhmdio** [comment]
👌 IMPROVE: remove provider.tf for more freedom

**Comment by sblack4** [comment]
I'll fix the pre-commit check but this looks good 👍  Thank you!

**Pull request merged**